Tensile/Source/lib/source/hip/HipSolutionAdapter.cpp

```diff
@@ -81,9 +82,27 @@ namespace Tensile
 
         hipError_t SolutionAdapter::loadCodeObjectFile(std::string const& path)
         {
-            hipModule_t module;
+            hipModule_t             module;
+            std::unique_ptr<char[]> buffer;
+            std::ifstream           coFile(path, std::ifstream::binary);
+
+            // hipModuleLoad holds the file descriptor/handle which can result in a process
+            // running out of descriptors/handles. Use hipModuleLoadData as a workaround
+            if(coFile)
+            {
+                coFile.seekg(0, coFile.end);
+                auto length = coFile.tellg();
+                coFile.seekg(0, coFile.beg);
 
-            HIP_CHECK_RETURN(hipModuleLoad(&module, path.c_str()));
+                buffer = std::make_unique<char[]>(length);
+                coFile.read(buffer.get(), length);
```

Comment:
What happens if any of these operations fail?

Reply:
If my understanding of `std::ifstream` is correct, these methods won't throw unless you set the `exceptions()` mask to tell it to. So if something goes wrong here and the buffer doesn't get filled with the correct data, we should still make it to the `hipModuleLoadData` call which will then return whatever hip error is relevant.